### DISCUSSION TITLE
Upgrading to v7, breaks my use of forwardRef

### DISCUSSION THREAD

**Discussion by davesag** [react-hook-form/react-hook-form#5110] [Q&A]
see also https://stackoverflow.com/questions/67463135/how-to-use-forwardref-in-a-render-prop-in-react-hook-form-v7

I'm trying to upgrade to v7 from v6 and replacing the `Controller` `as` with `render` is giving me issues.

I've outlined the issue in SatckOverflow per the link above and welcome any thoughts.

**Comment by bluebill1049** [comment]
```
export const DatePicker = styled(
  ({ name, validation, label, control, value, onChange, disablePast, ...props }) => (
    <Row id={name} {...props}>
      <Controller
        render={({ field }) =>
            <KeyboardDatePicker
              {...field}
              id={name}
              name={name}
              disableToolbar
              margin="dense"
              inputVariant="outlined"
              format="dd-MM-yyyy"
              label={label}
              value={value}
              onChange={onChange}
              autoOk
              inputRef={ref}
              disablePast={disablePast}
            />
        }
        name={name}
        control={control}
        rules={validation}
      />
    </Row>
  )
)(({ theme }) => ({
  padding: theme.spacing(0, 0, 1, 0)
}))
```

can you try without the forward ref, you don't need it, it should just worked.

**Reply by bluebill1049** [reply]
if not please provide a codesandbox.

**Reply by davesag** [reply]
alas no, the `forwardRef` is needed due to some weirdness with the `MuiDatePicker` - tried without and I get an error about refs and functional components (was one of the things I tried on the weekend).  We are looking into replacing `MuiDatePicker` with something less annoying.  I'll put a CSB together later this week.

**Reply by bluebill1049** [reply]
kk, will take a look the CSB.
